### DISCUSSION TITLE
syntax highlighting of the code

### DISCUSSION THREAD

**Discussion by LazarenkoA** [usememos/memos#3068] [General]
Hello, could you please tell me which component is responsible for code syntax highlighting? I mean ` ```yaml` or  ` ```go`, etc. I would like to create a pull request to make the syntax highlighting for the 1C programming language more authentic. I have grammars, and I even have syntax highlighting for Atom:

https://github.com/1c-syntax/atom-language-1c-bsl

**Comment by MehadND** [comment]
Hi try this file

web/src/components/MemoContent/CodeBlock.tsx

**Comment by LazarenkoA** [comment]
Can you tell me, when changing the syntax highlighting theme using the highlightjs library, will it affect the highlighting for all languages? (I would like to use the androidstudio theme.)

**Comment by MehadND** [comment]
Well I'm not really sure...haven't really did something like that before but I just checked highlight.js GitHub page for supported languages and it showed 1C in the list. 

About the AndroidStudio theme I'm not sure how to exactly do that bit. Sorry if that didn't help
